runtime/drivers/mysql/parser.go

```diff
@@ -0,0 +1,307 @@
+package mysql
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+	"strings"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+)
+
+type mapper interface {
+	runtimeType(st reflect.Type) (*runtimev1.Type, error)
+	// dest returns a pointer to a destination value that can be used in Rows.Scan
+	dest(st reflect.Type) (any, error)
+	// value dereferences a pointer created by dest
+	value(p any) (any, error)
+}
+
+// refer https://github.com/go-sql-driver/mysql/blob/master/fields.go for base types
+func getDBTypeNameToMapperMap() map[string]mapper {
+	m := make(map[string]mapper)
+
+	// bit
+	m["BIT"] = &bitMapper{}
+
+	// numeric
+	m["TINYINT"] = &numericMapper{}
+	m["SMALLINT"] = &numericMapper{}
+	m["MEDIUMINT"] = &numericMapper{}
+	m["INT"] = &numericMapper{}
+	m["UNSIGNED TINYINT"] = &numericMapper{}
+	m["UNSIGNED SMALLINT"] = &numericMapper{}
+	m["UNSIGNED INT"] = &numericMapper{}
+	m["UNSIGNED BIGINT"] = &numericMapper{}
+	m["BIGINT"] = &numericMapper{}
+	m["DOUBLE"] = &numericMapper{}
+	m["FLOAT"] = &numericMapper{}
+	// MySQL stores DECIMAL value in binary format
+	// It might be stored as string without losing precision
+	m["DECIMAL"] = &charMapper{}
+
+	// string
+	m["CHAR"] = &charMapper{}
+	m["LONGTEXT"] = &charMapper{}
+	m["MEDIUMTEXT"] = &charMapper{}
+	m["TEXT"] = &charMapper{}
+	m["TINYTEXT"] = &charMapper{}
+	m["VARCHAR"] = &charMapper{}
+
+	// binary
+	m["BINARY"] = &byteMapper{}
+	m["TINYBLOB"] = &byteMapper{}
+	m["BLOB"] = &byteMapper{}
+	m["LONGBLOB"] = &byteMapper{}
+	m["MEDIUMBLOB"] = &byteMapper{}
+	m["VARBINARY"] = &byteMapper{}
+
+	// date and time
+	m["DATE"] = &dateMapper{}
+	m["DATETIME"] = &dateMapper{}
+	m["TIMESTAMP"] = &dateMapper{}
+	m["YEAR"] = &numericMapper{}
+	// TIME is scanned as bytes and can be converted to string
+	m["TIME"] = &charMapper{}
+
+	// json
+	m["JSON"] = &jsonMapper{}
```

Comment:
Since these structs dont have state, I think they don't need to be pointers? For example, can just set `= bitMapper{}` instead of `= &bitMapper{}` (would need to change the member function signatures to work on non-pointer types)